lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php

```diff
@@ -123,41 +126,78 @@ public function convertException($message, DriverException $exception)
 
     /**
      * {@inheritdoc}
+     *
+     * @return AbstractPlatform|MariaDb102Platform|MySQL57Platform|MySqlPlatform
+     * @throws DBALException
      */
-    public function createDatabasePlatformForVersion($version)
+    public function createDatabasePlatformForVersion($version) : AbstractPlatform
```

Comment:
This might be a BC break, right? Since this is not an `@internal` class (at least is not marked with that)  people might inherit from it.

Reply:
Good catch ;)